### ISSUE TITLE
[Bug]: deadline exceed, historical query failed 

### ISSUE THREAD

**Issue by yovelcohen** [milvus-io/milvus#19818]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.1.4/2.1.2
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.1.3
- OS(Ubuntu or CentOS): CentOS
- CPU/Memory: 1 / 2GB

might be related:
https://github.com/milvus-io/milvus/issues/19399
https://github.com/milvus-io/milvus/issues/19618
https://github.com/milvus-io/milvus/issues/19027
```

### Current Behavior

I have a collection, roughly around 10k 768 dim vectors with no index.
suddenly querying (specifically calc_distance) has started to fail due to health issues in query node.
also getting similar errors in attu. 
tried downgrading to milvus v 2.1.2 and restarting the pods, but that didn't help.

### Expected Behavior

querying node should remain stable. 

### Milvus Log

logs from query node
```
{"level":"INFO","time":"2022/10/17 04:57:45.887 +00:00","caller":"querynode/shard_cluster.go:384","message":"ShardCluster remove segment","nodeID":254,"segmentID":436718046876270595,"state":0}

{"level":"WARN","time":"2022/10/17 04:57:45.887 +00:00","caller":"querynode/shard_cluster.go:396","message":"ShardCluster removeSegment found node not match","segmentID":436718046876270595,"nodeID":263,"evtNodeID":254}

{"level":"INFO","time":"2022/10/17 04:57:45.890 +00:00","caller":"querynode/shard_cluster_service.go:156","message":"successfully handoff","channel":"by-dev-rootcoord-dml_10_436647552388431873v0","segment":"online_segments:<segmentID:436718046876270595 collectionID:436647552388431873 partitionID:436647552388431874 num_rows:157 dmChannel:\"by-dev-rootcoord-dml_10_436647552388431873v0\" compactionFrom:436714113195835400 compactionFrom:436718034529026049 compactionFrom:436717999545909249 compactionFrom:436718044962357249 segment_state:Sealed replica_ids:436641986558296093 node_ids:263 > offline_segments:<segmentID:436718046876270595 collectionID:436647552388431873 partitionID:436647552388431874 num_rows:157 dmChannel:\"by-dev-rootcoord-dml_10_436647552388431873v0\" compactionFrom:436714113195835400 compactionFrom:436718034529026049 compactionFrom:436717999545909249 compactionFrom:436718044962357249 segment_state:Sealed replica_ids:436641986558296093 node_ids:254 > "}

{"level":"ERROR","time":"2022/10/17 04:58:22.191 +00:00","caller":"querynode/shard_cluster.go:886","message":"stream query failed on nodeID 250, err = deadline exceed","stack":"github.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Query.func1\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:886"}

{"level":"ERROR","time":"2022/10/17 04:58:22.191 +00:00","caller":"querynode/shard_cluster.go:910","message":"historical query failed on nodeID 250, reason = deadline exceed, err = %!w(<nil>)","stack":"github.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Query.func2\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:910"}

{"level":"WARN","time":"2022/10/17 04:58:22.191 +00:00","caller":"querynode/impl.go:652","message":"failed to query cluster","msgID":436727362562031665,"collectionID":436647551287427073,"error":"stream query failed on nodeID 250, err = deadline exceed, historical query failed on nodeID 250, reason = deadline exceed, err = %!w(<nil>)"}
```

### Anything else?
Maybe related: https://github.com/milvus-io/milvus/issues/18756
_No response_

**Comment by yovelcohen** [comment]
[milvus-logs.zip](https://github.com/milvus-io/milvus/files/9807812/milvus-logs.zip)
@yanliang567

**Comment by yovelcohen** [comment]
now noticing this log a lot as well

{"level":"ERROR","time":"2022/11/01 19:23:03.172 +00:00","caller":"querynode/shard_cluster.go:886","message":"stream query failed on nodeID 335, err = taskScheduler AddReadTask context is done","stack":"github.com/milvus-io/milvus/internal/querynode.(*ShardCluster).Query.func1\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:886"}

**Comment by yovelcohen** [comment]
also, 
```{"level":"INFO","time":"2022/11/04 11:00:10.045 +00:00","caller":"querycoord/task.go:1665","message":"start do handoff segments task","segmentIDs":[437115877278875650]}
{"level":"ERROR","time":"2022/11/04 11:00:10.048 +00:00","caller":"querycoord/task.go:1785","message":"handoffTask: handoff segment failed","segmentID":437115877278875650,"error":"sealed segment has been exist on query node, segmentID is 437115877278875650","stack":"github.com/milvus-io/milvus/internal/querycoord.(*handoffTask).execute\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task.go:1785\ngithub.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:593\ngithub.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).scheduleLoop\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:678"}
{"level":"WARN","time":"2022/11/04 11:00:10.048 +00:00","caller":"querycoord/task_scheduler.go:595","message":"failed to execute task","error":"sealed segment has been exist on query node, segmentID is 437115877278875650"}
{"level":"INFO","time":"2022/11/04 11:00:10.048 +00:00","caller":"querycoord/task.go:1800","message":"handoffTask postExecute done","taskID":437123594151626683}
{"level":"WARN","time":"2022/11/04 11:00:10.048 +00:00","caller":"querycoord/task_scheduler.go:706","message":"scheduleLoop: process triggerTask failed","triggerTaskID":437123594151626683,"error":"sealed segment has been exist on query node, segmentID is 437115877278875650"}
{"level":"INFO","time":"2022/11/04 11:00:10.048 +00:00","caller":"querycoord/task_scheduler.go:746","message":"scheduleLoop: start rollBack after triggerTask failed","triggerTaskID":437123594151626683,"rollBackTasks":[],"error":", sealed segment has been exist on query node, segmentID is 437115877278875650"}
{"level":"WARN","time":"2022/11/04 11:00:10.048 +00:00","caller":"querycoord/handoff_handler.go:391","message":"HandoffHandler: handoff task failed to execute","segmentID":437115877278875650,"error":", sealed segment has been exist on query node, segmentID is 437115877278875650"}
{"level":"INFO","time":"2022/11/04 11:00:10.052 +00:00","caller":"querycoord/task_scheduler.go:761","message":"scheduleLoop: trigger task done and delete from etcd","triggerTaskID":437123594151626683}
{"level":"INFO","time":"2022/11/04 11:00:10.052 +00:00","caller":"querycoord/task_scheduler.go:675","message":"scheduleLoop: pop a triggerTask from triggerTaskQueue","triggerTaskID":437123594151626684}
{"level":"INFO","time":"2022/11/04 11:00:10.052 +00:00","caller":"querycoord/task_scheduler.go:505","message":"begin to process task","taskID":437123594151626684,"task":"*querycoord.handoffTask"}
{"level":"INFO","time":"2022/11/04 11:00:10.052 +00:00","caller":"querycoord/task.go:1665","message":"start do handoff segments task","segmentIDs":[437115876282728449]}
{"level":"ERROR","time":"2022/11/04 11:00:10.055 +00:00","caller":"querycoord/task.go:1785","message":"handoffTask: handoff segment failed","segmentID":437115876282728449,"error":"sealed segment has been exist on query node, segmentID is 437115876282728449","stack":"github.com/milvus-io/milvus/internal/querycoord.(*handoffTask).execute\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task.go:1785\ngithub.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:593\ngithub.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).scheduleLoop\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:678"}
{"level":"WARN","time":"2022/11/04 11:00:10.055 +00:00","caller":"querycoord/task_scheduler.go:595","message":"failed to execute task","error":"sealed segment has been exist on query node, segmentID is 437115876282728449"}
{"level":"INFO","time":"2022/11/04 11:00:10.055 +00:00","caller":"querycoord/task.go:1800","message":"handoffTask postExecute done","taskID":437123594151626684}
{"level":"WARN","time":"2022/11/04 11:00:10.055 +00:00","caller":"querycoord/task_scheduler.go:706","message":"scheduleLoop: process triggerTask failed","triggerTaskID":437123594151626684,"error":"sealed segment has been exist on query node, segmentID is 437115876282728449"}
{"level":"INFO","time":"2022/11/04 11:00:10.055 +00:00","caller":"querycoord/task_scheduler.go:746","message":"scheduleLoop: start rollBack after triggerTask failed","triggerTaskID":437123594151626684,"rollBackTasks":[],"error":", sealed segment has been exist on query node, segmentID is 437115876282728449"}
{"level":"WARN","time":"2022/11/04 11:00:10.055 +00:00","caller":"querycoord/handoff_handler.go:391","message":"HandoffHandler: handoff task failed to execute","segmentID":437115876282728449,"error":", sealed segment has been exist on query node, segmentID is 437115876282728449"}
{"level":"INFO","time":"2022/11/04 11:00:10.058 +00:00","caller":"querycoord/task_scheduler.go:761","message":"scheduleLoop: trigger task done and delete from etcd","triggerTaskID":437123594151626684}
{"level":"INFO","time":"2022/11/04 11:00:10.058 +00:00","caller":"querycoord/task_scheduler.go:675","message":"scheduleLoop: pop a triggerTask from triggerTaskQueue","triggerTaskID":437123594151626685}
{"level":"INFO","time":"2022/11/04 11:00:10.058 +00:00","caller":"querycoord/task_scheduler.go:505","message":"begin to process task","taskID":437123594151626685,"task":"*querycoord.handoffTask"}
{"level":"INFO","time":"2022/11/04 11:00:10.058 +00:00","caller":"querycoord/task.go:1665","message":"start do handoff segments task","segmentIDs":[437115876492443649]}
{"level":"ERROR","time":"2022/11/04 11:00:10.062 +00:00","caller":"querycoord/task.go:1785","message":"handoffTask: handoff segment failed","segmentID":437115876492443649,"error":"sealed segment has been exist on query node, segmentID is 437115876492443649","stack":"github.com/milvus-io/milvus/internal/querycoord.(*handoffTask).execute\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task.go:1785\ngithub.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:593\ngithub.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).scheduleLoop\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:678"}
{"level":"WARN","time":"2022/11/04 11:00:10.062 +00:00","caller":"querycoord/task_scheduler.go:595","message":"failed to execute task","error":"sealed segment has been exist on query node, segmentID is 437115876492443649"}
{"level":"INFO","time":"2022/11/04 11:00:10.062 +00:00","caller":"querycoord/task.go:1800","message":"handoffTask postExecute done","taskID":437123594151626685}
{"level":"WARN","time":"2022/11/04 11:00:10.062 +00:00","caller":"querycoord/task_scheduler.go:706","message":"scheduleLoop: process triggerTask failed","triggerTaskID":437123594151626685,"error":"sealed segment has been exist on query node, segmentID is 437115876492443649"}
{"level":"INFO","time":"2022/11/04 11:00:10.062 +00:00","caller":"querycoord/task_scheduler.go:746","message":"scheduleLoop: start rollBack after triggerTask failed","triggerTaskID":437123594151626685,"rollBackTasks":[],"error":", sealed segment has been exist on query node, segmentID is 437115876492443649"}
{"level":"WARN","time":"2022/11/04 11:00:10.062 +00:00","caller":"querycoord/handoff_handler.go:391","message":"HandoffHandler: handoff task failed to execute","segmentID":437115876492443649,"error":", sealed segment has been exist on query node, segmentID is 437115876492443649"}
{"level":"INFO","time":"2022/11/04 11:00:10.065 +00:00","caller":"querycoord/task_scheduler.go:761","message":"scheduleLoop: trigger task done and delete from etcd","triggerTaskID":437123594151626685}
{"level":"INFO","time":"2022/11/04 11:00:14.821 +00:00","caller":"querycoord/handoff_handler.go:239","message":"handoff task scheduled: ","task number":47}
{"level":"ERROR","time":"2022/11/04 11:00:14.822 +00:00","caller":"querycoord/global_meta_broker.go:514","message":"QueryCoord acquire the segment reference lock error","segIDs":[437115961046990849],"failed reason":"segment is not exist with ID = 437115961046990849","stack":"github.com/milvus-io/milvus/internal/querycoord.(*globalMetaBroker).acquireSegmentsReferLock\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/global_meta_broker.go:514\ngithub.com/milvus-io/milvus/internal/querycoord.(*HandoffHandler).process\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/handoff_handler.go:270\ngithub.com/milvus-io/milvus/internal/querycoord.(*HandoffHandler).schedule\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/handoff_handler.go:241"}
{"level":"WARN","time":"2022/11/04 11:00:14.823 +00:00","caller":"querycoord/handoff_handler.go:275","message":"HandoffHandler: acquire segment reference lock failed","segmentID":437115961046990849,"error":"segment is not exist with ID = 437115961046990849"}
{"level":"INFO","time":"2022/11/04 11:00:14.823 +00:00","caller":"querycoord/handoff_handler.go:353","message":"HandoffHandler: trigger handoff","segmentInfo":"segmentID:437116756164870146 collectionID:437009147787476993 partitionID:437009147787476994 dmChannel:\"by-dev-rootcoord-dml_36_437009147787476993v0\" compactionFrom:437115964966305793 compactionFrom:437009556876296193 compactionFrom:437105217798144001 compactionFrom:437009541462228993 compactionFrom:437105214639308803 compactionFrom:437102452834304003 compactionFrom:437101640397357061 compactionFrom:437101225515417604 createdByCompaction:true segment_state:Sealed "}
{"level":"INFO","time":"2022/11/04 11:00:14.833 +00:00","caller":"querycoord/handoff_handler.go:382","message":"HandoffHandler: handoff task triggered successfully","segmentID":437116756164870146}
{"level":"INFO","time":"2022/11/04 11:00:14.833 +00:00","caller":"querycoord/handoff_handler.go:353","message":"HandoffHandler: trigger handoff","segmentInfo":"segmentID:437115793851547650 collectionID:437009158286606337 partitionID:437009158286606338 dmChannel:\"by-dev-rootcoord-dml_43_437009158286606337v1\" segment_state:Sealed "}
{"level":"INFO","time":"2022/11/04 11:00:14.833 +00:00","caller":"querycoord/task_scheduler.go:675","message":"scheduleLoop: pop a triggerTask from triggerTaskQueue","triggerTaskID":437123594151626686}
{"level":"INFO","time":"2022/11/04 11:00:14.833 +00:00","caller":"querycoord/task_scheduler.go:505","message":"begin to process task","taskID":437123594151626686,"task":"*querycoord.handoffTask"}
{"level":"INFO","time":"2022/11/04 11:00:14.833 +00:00","caller":"querycoord/task.go:1665","message":"start do handoff segments task","segmentIDs":[437116756164870146]}
{"level":"INFO","time":"2022/11/04 11:00:14.840 +00:00","caller":"querycoord/handoff_handler.go:382","message":"HandoffHandler: handoff task triggered successfully","segmentID":437115793851547650}
{"level":"ERROR","time":"2022/11/04 11:00:14.840 +00:00","caller":"querycoord/task.go:1785","message":"handoffTask: handoff segment failed","segmentID":437116756164870146,"error":"sealed segment has been exist on query node, segmentID is 437116756164870146","stack":"github.com/milvus-io/milvus/internal/querycoord.(*handoffTask).execute\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task.go:1785\ngithub.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:593\ngithub.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).scheduleLoop\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:678"}``

**Comment by yhmo** [comment]
Another user also encountered this problem:
[milvus-querycoord-76bdffcc98-mkjnp.log](https://github.com/milvus-io/milvus/files/9958996/milvus-querycoord-76bdffcc98-mkjnp.log)

2000 entities put into a collection. Load the collection, unload the collection, try to load the entire collection and it hangs

**Comment by xiaofan-luan** [comment]
hi @yovelcohen , this might not be a issue in 2.2.0 because the in the new release we entirely refactor the querycoord.
If you could I would suggest to upgrade to 2.2 directly

**Comment by xiaofan-luan** [comment]
If the cluster still stucked for some reason, try to release the collection (if failed to force release with bird watcher) might help.

**Comment by yovelcohen** [comment]
Thanks for the response @xiaofan-luan , appreciate it. 
we plan to upgrade to 2.2 but right now the migration from 2.1.2 did not work for us at all.

**Comment by yah01** [comment]
Hi @yovelcohen 
If you are planning to upgrade, could tell what the problem is while migration? 
or stay in v2.1, you could use [birdwatcher](https://github.com/milvus-io/birdwatcher) to force release the collection, this could help. Please make sure to follow these steps:
- Stop milvus, but keep at least etcd alive
- Run birdwatcher to connect to the etcd
- Run force-release command
- Restart milvus

**Comment by yovelcohen** [comment]
yep, those steps helps, I'm able to get the query nodes up and running again after that.  
But, I can't really state that milvus is production ready if I have to manually fix those once every few days/hours... 
I did see in another bug a small guide on how to migrate to 2.2, we'll try that and hopefully it'll be fixed

**Comment by yanliang567** [comment]
@yovelcohen any updates on milvus 2.2?
